mlflow/tensorflow/__init__.py

```diff
@@ -1257,6 +1265,29 @@ def __restore_generator(prev_generator):
                 if log_models:
                     _log_keras_model(history, args)
 
+                if log_datasets:
+                    try:
+                        # create a CodeDatasetSource
+                        context_tags = context_registry.resolve_tags()
+                        source = CodeDatasetSource(
+                            mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                            mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                        )
+
+                        training_data = kwargs["x"] if "x" in kwargs else args[0]
+                        # create a dataset
+                        if isinstance(training_data, np.ndarray):
+                            dataset = from_numpy(features=training_data, source=source)
+                            mlflow.log_input(dataset, "train")
+                        elif isinstance(training_data, tensorflow.Tensor):
+                            dataset = from_tensorflow(data=training_data, source=source)
+                            mlflow.log_input(dataset, "train")
+                        elif isinstance(training_data, tensorflow.data.Dataset):
```

Comment:
Can we add an `else` case and print a warning if the data is of an unrecognized type?